lua/terrortown/menus/gamemode/base_gamemodemenu.lua

```diff
@@ -102,6 +106,61 @@ function CLGAMEMODEMENU:AddSubmenu(submenu)
 	self.submenus[#self.submenus + 1] = submenu
 end
 
+---
+-- Checks if the menu has a searchbar enabled.
+-- @note This function should be overwritten and return true, if you want a searchbar.
+-- @return boolean Return true if searchbar should be available
+-- @realm client
+function CLGAMEMODEMENU:HasSearchbar()
+	return false
+end
+
+---
+-- Filters the list with a searchText and returns full list if nothing is entered.
+-- @note Overwrite MatchesSearchString for a custom search! 
+-- This function can be overwritten, but probably shouldnt. 
+-- @param string searchText
+-- @return menuClasses Returns a list of all matching submenus, needs to be indexed with ascending numbers
+-- @realm client
+function CLGAMEMODEMENU:GetMatchingSubmenus(searchText)
+	local submenuClasses = self:GetVisibleSubmenus()
+
+	if searchText == "" then
+		return submenuClasses
+	end
+
+	local filteredSubmenuClasses = {}
+
+	local counter = 0
+	for i = 1, #submenuClasses do
+		local submenuClass = submenuClasses[i]
+
+		if self:MatchesSearchString(submenuClass, searchText) then
+			counter = counter + 1
+			filteredSubmenuClasses[counter] = submenuClass
+		end
+	end
+
+	return filteredSubmenuClasses
+end
+
+---
+-- Determines the used searchfunction, when a HasSearchbar returns true.
+-- Parameters for that function are submenuClasses and the searchText
+-- Per default only titles are searched and compared to the searchtext in lowercase letters.
+-- @note This function can be overwritten to use a custom searchfunction.
+-- @param menuClass submenuClass
+-- @param string searchText
+-- @return bool Returns 
+-- @realm client
+function CLGAMEMODEMENU:MatchesSearchString(submenuClass, searchText)
+	local txt = stringLower(searchText)
+	local title = stringLower(TryT(submenuClass.title))
+	local start = stringFind(title, txt)
+
+	return start or false and true
```

Comment:
Regarding <https://wiki.facepunch.com/gmod/string.find>, a number is returned if found.
Regarding <https://wiki.facepunch.com/gmod/Global.tobool>, tobool() should work, so `tobool(start)` or `start ~= nil`.